test/e2e/kubevirt.go

```diff
@@ -347,6 +262,59 @@ passwd:
 			return fr.ClientSet.NetworkingV1().NetworkPolicies(namespace).Create(context.TODO(), policy, metav1.CreateOptions{})
 		}
 
+		checkEastWestTraffic = func(vmi *kubevirtv1.VirtualMachineInstance, podIPsByName map[string][]string, stage string) {
+			polling := time.Second
+			timeout := 10 * time.Second
+			for podName, podIPs := range podIPsByName {
+				for _, podIP := range podIPs {
+					output := ""
+					Eventually(func() error {
+						var err error
+						output, err = kubevirt.RunCommand(vmi, fmt.Sprintf("curl http://%s", net.JoinHostPort(podIP, "8000")), polling)
+						return err
+					}).
+						WithOffset(1).
+						WithPolling(polling).
+						WithTimeout(timeout).
+						Should(Succeed(), func() string { return stage + ": " + podName + ": " + output })
+				}
+			}
+		}
+
+		httpServerTestPodsDefaultNetworkIPs = func() map[string][]string {
+			ips := map[string][]string{}
+			for _, pod := range httpServerTestPods {
+				for _, podIP := range pod.Status.PodIPs {
+					ips[pod.Name] = append(ips[pod.Name], podIP.IP)
+				}
+			}
+			return ips
+		}
+
+		checkPodHasIPsAtNetwork = func(netName string, expectedNumberOfAddresses int) func(Gomega, *corev1.Pod) {
+			return func(g Gomega, pod *corev1.Pod) {
```

Comment:
having to receive gomega is avoidable; I would rather have this function return an error, and assert in the caller.

Reply:
This is better since you will get a proper matching error 
```golang
g.Expect(err).ToNot(HaveOccurred())                             
g.Expect(netStatus).To(HaveLen(1))                              
g.Expect(netStatus[0].IPs).To(HaveLen(expectedNumberOfAddresses))
```

Instead of a custom error you have to manually compose to be descriptive of the type of failure.

Reply:
might be just me, but I unfortunately don't follow why what you're proposing is better.